jddd-architecture/jddd-architecture-onion/src/main/java/org/jddd/architecture/onion/ApplicationServiceRing.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jddd.architecture.onion;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Identifies the {@link ApplicationServiceRing} in an onion architecture. The application service ring implements and
+ * orchestrates business use case. To do so, it only depends on the inner rings, i.e. {@link DomainModelRing} and
+ * {@link DomainServiceRing}.
+ *
+ * @author Christian Stettler
+ * @author Henning Schwendtner
+ * @author Stephan Pirnbaum
+ * @author Martin Schimak
+ * @author Oliver Drotbohm
+ * @see <a href="https://jeffreypalermo.com/2008/07/the-onion-architecture-part-1/">The Onion Architecture : part 1
+ * (Palermo)</a>
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target({ ElementType.PACKAGE, ElementType.TYPE })
```

Comment:
What would be the use case for using the `@ApplicationServiceRing` stereotype on a type level (same question for the other stereotypes)?

Reply:
Use case would be for identifying classes in an existing code base which is not yet structured with packages/modules, i.e. when restructuring an application using the onion architecture

Reply:
Agreed. The technical structuring (here: rings) might not be reflected on the package level in the first place, as that has implications on the required visibility of types which might not be desired for the particular project. In this case the semantics of the annotation are "The annotated type belongs to the…".